### ISSUE TITLE
WordPress files get removed with Composer v2

### ISSUE THREAD

**Issue by szepeviktor** [johnpbloch/wordpress-core-installer#29]
```json
{
    "require": {
        "johnpbloch/wordpress": "^5.5"
    }
}
```

then issue `composer update -vvv`

```
Running 2.0.5 (2020-11-06 20:57:15) with PHP 7.4.12 on Linux / 4.9.0-13-amd64
Loading composer repositories with package information
Updating dependencies
Downloading https://repo.packagist.org/packages.json
[200] https://repo.packagist.org/packages.json
Writing /home/viktor/.composer/cache/repo/https---repo.packagist.org/packages.json into cache
Reading /home/viktor/.composer/cache/repo/https---repo.packagist.org/provider-johnpbloch~wordpress.json from cache
Downloading https://repo.packagist.org/p2/johnpbloch/wordpress.json if modified
[304] https://repo.packagist.org/p2/johnpbloch/wordpress.json
Reading /home/viktor/.composer/cache/repo/https---repo.packagist.org/provider-johnpbloch~wordpress-core-installer.json from cache
Downloading https://repo.packagist.org/p2/johnpbloch/wordpress-core-installer.json if modified
Reading /home/viktor/.composer/cache/repo/https---repo.packagist.org/provider-johnpbloch~wordpress-core.json from cache
Downloading https://repo.packagist.org/p2/johnpbloch/wordpress-core.json if modified
[304] https://repo.packagist.org/p2/johnpbloch/wordpress-core-installer.json
[304] https://repo.packagist.org/p2/johnpbloch/wordpress-core.json
Generating rules
Resolving dependencies through SAT
Looking at all rules.

Dependency resolution completed in 0.001 seconds
Analyzed 102 packages to resolve dependencies
Analyzed 119 rules to resolve dependencies
Lock file operations: 3 installs, 0 updates, 0 removals
Installs: johnpbloch/wordpress-core-installer:2.0.0, johnpbloch/wordpress-core:5.5.3, johnpbloch/wordpress:5.5.3
  - Locking johnpbloch/wordpress (5.5.3)
  - Locking johnpbloch/wordpress-core (5.5.3)
  - Locking johnpbloch/wordpress-core-installer (2.0.0)
Writing lock file
Installing dependencies from lock file (including require-dev)
Reading ./composer.lock
Package operations: 3 installs, 0 updates, 0 removals
Installs: johnpbloch/wordpress-core-installer:2.0.0, johnpbloch/wordpress-core:5.5.3, johnpbloch/wordpress:5.5.3
Reading /home/viktor/.composer/cache/files/johnpbloch/wordpress-core-installer/84ce93917e6c7c835605347db4d7b24a35c31b36.zip from cache
  - Loading johnpbloch/wordpress-core-installer (2.0.0) from cache
Reading /home/viktor/.composer/cache/files/johnpbloch/wordpress-core/e85f10502a0a6f4864337a4474036d50e388215f.zip from cache
  - Loading johnpbloch/wordpress-core (5.5.3) from cache
Reading /home/viktor/.composer/cache/files/johnpbloch/wordpress/b8fa87abf133edd70fc2c40d173a209a2ca02d61.zip from cache
  - Loading johnpbloch/wordpress (5.5.3) from cache
  - Installing johnpbloch/wordpress-core-installer (2.0.0): Extracting archive
Executing async command (CWD): unzip -qq  '/home/viktor/src/aaa/vendor/composer/tmp-489c3d55a29081837b520ea3a4b8f684' -d '/home/viktor/src/aaa/vendor/composer/e7189ce9'
Executing command (CWD): rm -rf '/home/viktor/src/aaa/vendor/johnpbloch/wordpress-core-installer'
Executing command (CWD): rm -rf '/home/viktor/src/aaa/vendor/composer/e7189ce9'
Loading plugin johnpbloch\Composer\WordPressCorePlugin
  - Installing johnpbloch/wordpress-core (5.5.3): Extracting archive
Executing async command (CWD): unzip -qq  '/home/viktor/src/aaa/vendor/composer/tmp-27df48c3e76ad610b4feb1a57b7ff83c' -d '/home/viktor/src/aaa/vendor/composer/9ab5b618'
  - Installing johnpbloch/wordpress (5.5.3): Extracting archive
Executing async command (CWD): unzip -qq  '/home/viktor/src/aaa/vendor/composer/tmp-186e3fe3d05bdc276f37ac0b34690add' -d '/home/viktor/src/aaa/vendor/composer/6ddaa97c'
Executing command (CWD): rm -rf '/home/viktor/src/aaa/vendor/johnpbloch/wordpress'
Executing command (CWD): rm -rf '/home/viktor/src/aaa/vendor/composer/6ddaa97c'
Executing command (CWD): rm -rf 'wordpress'
Executing command (CWD): rm -rf '/home/viktor/src/aaa/vendor/composer/9ab5b618'
Executing command (CWD): rm -rf '/home/viktor/src/aaa/wordpress'
Executing command (CWD): rm -rf '/home/viktor/src/aaa/vendor/johnpbloch/wordpress-core'
Downloading https://packagist.org/downloads/
[201] https://packagist.org/downloads/
Generating autoload files
```

So WP files got removed.

Works with Composer v1.

@johnpbloch Please advise.

**Comment by szepeviktor** [comment]
👀 https://github.com/composer/installers/issues/467#issuecomment-724091848

Quote from the ["Principles"](https://gist.github.com/szepeviktor/6c5b75f84e5c999eb8232cd963c8e6f5)

> Do not install new major version releases of any software in production, let others test it first

**Comment by tomjn** [comment]
@szepeviktor I tried to reproduce the removal of the folder and could see no erroneous behaviour. Composer is meant to delete and replace package folders.

The only way I can see this being a problem is:

 - if you've made modifications to WP core
 - if you've tried to make a classic WP install with `wordpress/wp-content/etc...`

The solution to both of which is... ***don't***.

**Comment by szepeviktor** [comment]
@tomjn As linked above Composer 2.0.5 had a bug that cause all these.
It was fixed in 2.0.6!
